src/com/shmuelzon/HomeAssistantFloorPlan/Controller.java

```diff
@@ -892,9 +892,12 @@ private void generateSensorEntities(Map<String, Entity> homeAssistantEntities) {
     private boolean doStateIconsIntersect(Entity first, Entity second) {
         final double STATE_ICON_RADIUS_INCLUDING_MARGIN = 20.0;
 
-        double x = Math.pow(first.position.x - second.position.x, 2) + Math.pow(first.position.y - second.position.y, 2);
+        double centerDistance = Math.sqrt(
+                Math.pow(first.position.x - second.position.x, 2) + Math.pow(first.position.y - second.position.y, 2)
+        );
 
-        return x <= Math.pow(STATE_ICON_RADIUS_INCLUDING_MARGIN * 2, 2);
+        // Icons intersect or touching each-other?
```

Comment:
I feel this comment is redundant. The entire name of the function is `doStateIconsIntersect`.

Reply:
Actually, intersection check has the operator `<` while touching has operator `==`.
See https://www.geeksforgeeks.org/check-two-given-circles-touch-intersect/

To prevent refactoring the method's name, I felt like adding a comment would be helpful for people who want to contribute.

IMHO, documenting (javadoc) and commenting code is very important to keep the code understandable. Certainly in projects where multiple people can maintain the code, like open-source projects. Comments and such are removed when compiling, anyway.

Reply:
`<` vs `==` is a bit pedantic, even for me. The only reason `=` is there is so we don't, unnecessarily, increase the distance between two icons if we don't have to but the purpose of the function is still the same: do they two icons intersect and, if so, we need to distance them from each other.

Javadoc is great for documenting public APIs for other users to make use of. Seems less important to me here. Generally speaking, I'd prefer the code to document itself with proper naming for functions and variables, where it's important. Too many comments just add noise and possible comment rot.